CHANGELOG.md

```diff
@@ -3,6 +3,64 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.0.1
+
+_Nov 17, 2021_
+
+A big thanks to the 3 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 New API to validate the editing values (#3006) @m4theushw
+ 
+  You can now use the `preProcessEditCellProps` key in `GridColDef` to synchronously or asynchronously validate the values committed.
+
+  ```ts
+  const columns: GridColDef[] = [
+    {
+      field: 'firstName',
+      preProcessEditCellProps: (params: GridEditCellPropsChangeParams) => {
+        const hasError = params.props.value.length < 3;
+        return { ...params.props, error: hasError };
+      },
+    },
+    {
+      field: 'email',
+      preProcessEditCellProps: async (params: GridEditCellPropsChangeParams) => {
+        const userWithEmail = await fetchUserByEmail(params.value);
+        const hasError = !!userWithEmail;
+        return { ...params.props, error: hasError };
+      }
+    }
+  ];
+  ```
+
+- ✨ New method `getRootDimensions` to access the size of the grid (#3007) @flaviendelangle
+
+  It contains the size of the viewport (which is the scrollable container containing the rows and columns) considering scrollbars or not.
+
+  ```ts
+  const dimensions = apiRef.current.getRootDimensions();
+  ```
+
+### `@mui/x-data-grid@v5.0.1` / `@mui/x-data-grid-pro@v5.0.1`
+
+#### Changes
+
+- [DataGrid] New API to validate the editing values (#3006) @m4theushw
+- [DataGrid] Use color-scheme to set dark mode on native components (#3146) @alexfauquette
+- [DataGrid] Fix the `@mui/x-data-grid` type entrypoint (#3196) @flaviendelangle
+
+### Core
```

Comment:
I think that "Docs" should come before "Core". I think that changes in "Core" are most interesting for maintainers.